Bindings/Java/OpenSimJNI/OpenSimContext.cpp

```diff
@@ -432,4 +432,21 @@ void OpenSimContext::restoreStateFromCachedModel()
     this->setState(&(_model->updWorkingState()));
     this->realizePosition();
 }
+
+void OpenSimContext::setConnecteeName(Component& comp, AbstractSocket& socket, const std::string& newValue) {
+    // Since some socket changes can invalidate the socket altogether we'll do the change conservatively by
+    // 1. Making clone of model and State
+    // 2. Find corresponding comp, socket in clone, apply the change
+    // 3. If successful, change is safe, redo on Model and be done
+    // 4. If it fails exception is thrown and we return, model is unchanged
+    cacheModelAndState();
+    clonedModel->initSystem(); // needed to populate socket/connection in clone
```

Comment:
I actually didn't have it initially and ended up woth a null pointer on the following line since cloning by itself doesn't create connections/sockets. I can create a test case to demonstrate if needed.

Reply:
Since these are connections, I'd assume finalizeConnections would work. The call line will look like:
'clonedModel->finalizeConnections(*clonedModel);'
I understand this could be fatser but to me looks much more confusing. 
Your call.

Reply:
> I actually didn't have it initially and ended up with a null pointer on the following line since cloning by itself doesn't create connections/sockets

What is returning a nullptr? Nothing in the following lines should require finalizing connections AFAICT. You are only trying to access a Socket, right? Sockets are part of the Component and they are not added by any later process.

Talking about confusing: what is `cacheModelAndState` actually doing? Isn't the clonedModel already "live" or does it make a clone right there and then?  If `clonedModel->updComponent()` is throwing (a nullptr is not returned) that suggests that the `clonedModel`'s properties are not up-to-date and the ownership tree has not been made. That tells me you didn't use `model->clone()`, which would `finalizeFromProperties()` for you.

It seems strange to me to clone the model on every edit. It seems more efficient to maintain a `clonedModel` (e.g. cached model. It would be helpful to stick to one naming convention) upon which edit operations are performed on the cloned and `initSystem` is called on it. If `initSystem` succeeds then `model = clonedModel` (this could be a clone or could transfer the pointer, whatever) and if it fails you relay message to user and delete `clonedModel` and restore `clonedModel = model.clone()`. Otherwise I don't really understand why the Context has/requires `clonedModel` as a member.

Reply:
@aseth1 the reference to the socket was nullptr since the cloned model is just a clone and initSystem was not invoked on it, it is kept just in case. As per our discussion, the cloning is done before the edit by the GUI because it needs to edit actual Properties that are referenced by UI, we can do edits on clone first and keep it in sync (and would be faster) but that requires all edits to go through a custom workflow rather than using Property edits. We agreed that blind cloning can be avoided if we have better granularity over what edits can change: Visualization, State, Topology

Reply:
> the reference to the socket was nullptr since the cloned model is just a clone and initSystem was not invoked on it, it is kept just in case

This shouldn't be the case. You should be able to access a component's Socket and it should throw if it does not have a Socket of the given name. There is either a bug or there is more going on that a simple clone.

Reply:
@aseth1 as indicated, this was not working for me and I thought we agreed to investigate it separately, if you'd like you can remove the call to initSystem, build the code in debug mode and see where/why it fails.

Reply:
I tried it. It does not fail. I am quite certain the issue was something else in the process. I will push changes to  this PR.